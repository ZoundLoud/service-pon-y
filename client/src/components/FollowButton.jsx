import React from 'react';

class FollowButton extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isFollowing: this.props.isFollowing,
    };
    this.toggleFollow = this.toggleFollow.bind(this);
  }

  toggleFollow() {
    const { isFollowing } = this.state;
    this.setState({
      isFollowing: !isFollowing,
    });
  }

  render() {
    const { isFollowing } = this.state;
    const followToolTip = isFollowing ? 'Unfollow' : 'Follow';
    if (isFollowing) {
      return (
        <button id="followingArtistButton" onClick={this.toggleFollow} title={followToolTip}>Following</button>
      );
    }
    return (
      <button id="followArtistButton" onClick={this.toggleFollow} title={followToolTip}>Follow</button>
    );
  }
}

export default FollowButton;
