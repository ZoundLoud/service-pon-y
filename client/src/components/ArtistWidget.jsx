import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faUserFriends, faHeadphones, faCircle, faStar, faSquare, faExclamation,
} from '@fortawesome/free-solid-svg-icons';
import NumericLabel from 'react-pretty-numbers';
import FollowButton from './FollowButton';

const params = {
  justification: 'L',
  locales: 'en-AU',
  currency: false,
  percentage: false,
  precision: 1,
  commafy: false,
  shortFormat: true,
  shortFormatMinValue: 1001,
  title: false,
};

function ArtistWidget({ artistData }) {
  if (!artistData) {
    return null;
  }

  const {
    avatar_picture, artist_name, no_of_followers, no_of_tracks, is_followed,
  } = artistData;
  return (
    <div className="artistWidget">
      <img src={avatar_picture} alt="avatar" />
      <div id="artistName">
        <span className="artistNameToolTipContainer" title={`Visit ${artist_name}'s Profile`}>
          {artist_name}
        </span>
        <span className="fa-layers fa-fw" style={{ marginLeft: '3px' }}>
          <span style={{ fontSize: '10px' }}>
            <FontAwesomeIcon icon={faCircle} size="m" color="#f50" />
          </span>
          <span style={{ fontSize: '8px' }}>
            <FontAwesomeIcon icon={faStar} size="xs" color="#fff" />
          </span>
        </span>

      </div>
      <div className="followAndTrackCount">
        <span className="ArtistFollowBadge" id={no_of_followers} title={`${no_of_followers.toLocaleString()} followers`}>
          <span style={{ marginRight: '3px' }}>
            <FontAwesomeIcon color="#999" icon={faUserFriends} />
          </span>
          <NumericLabel params={params}>{no_of_followers}</NumericLabel>
        </span>
        <span className="ArtistTrackBadge" title={`${no_of_tracks} tracks`}>
          <span style={{ marginRight: '3px' }}>
            <FontAwesomeIcon color="#666" padding-right="3px" icon={faHeadphones} />
          </span>
          {no_of_tracks}
        </span>

      </div>
      <FollowButton isFollowing={is_followed} />
      <div className="reportButton" style={{ marginTop: '30px' }}>
        <span className="fa-layers fa-fw">
          <FontAwesomeIcon icon={faSquare} size="1x" transform={{ rotate: 45 }} />
          <span style={{ fontSize: '7px' }}>
            <FontAwesomeIcon icon={faExclamation} size="1x" color="#fff" />
          </span>
        </span>
        {'  Report'}
      </div>
    </div>

  );
}

export default ArtistWidget;
