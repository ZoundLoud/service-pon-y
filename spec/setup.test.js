import React from 'react';
import Enzyme from 'enzyme';
import { mount, shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ArtistWidget from '../client/src/components/ArtistWidget';
import FollowButton from '../client/src/components/FollowButton';

Enzyme.configure({ adapter: new Adapter() });

describe('ArtistWidget', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ArtistWidget debug />);

    expect(component).toMatchSnapshot();
  });
});

it('should be possible to click Follow Button', () => {
  const component = mount(<FollowButton />);
  component
    .find('button')
    .simulate('click');
  expect(component).toMatchSnapshot();
  component.unmount();
});
