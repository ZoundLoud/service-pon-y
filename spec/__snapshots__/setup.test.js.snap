// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistWidget should render correctly in "debug" mode 1`] = `""`;

exports[`should be possible to click Follow Button 1`] = `
<FollowButton>
  <button
    id="followingArtistButton"
    onClick={[Function]}
    title="Unfollow"
  >
    Following
  </button>
</FollowButton>
`;
